src/main/webapp/scripts/script.js

```diff
@@ -0,0 +1,23 @@
+const API_KEY = 'AIzaSyAr7MKPZKUCQ9G-N00I44EJqvsYjE3jAqU';
+const YOUTUBE_DISCOVERY = 'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest';
+
+function loadApi(callback) {
+  gapi.load('client', function() {
+    gapi.client.init({
+      'apiKey': API_KEY,
+      'discoveryDocs': [YOUTUBE_DISCOVERY]
+    })
+    .then(function() {
+      callback();
+    })
+    .catch((error) => {
+      updateDom('Oops! An error occured with Youtube, please try again later.', 'popular-list-container');
+      console.log('Was not able to load Google API client:', error);
```

Comment:
Instead of logging just throw the error after updating the DOM. If someone else modified this file and added more code after gapi.load in this method, it would continue to execute even after you catch the error. This can introduce bugs when the next author doesn't notice this.